src/lTerm_widget.mli

```diff
@@ -33,6 +33,13 @@ class t : string -> object
   method can_focus : bool
     (** Whether the widget can receive the focus or not. *)
 
+  method focus : (t option * t option * t option * t option) option
```

Comment:
I think it would be slightly clearer if the various directions where named by using a record. It'd be complicated to refer to `t` in the record, but we could have something like this:

```ocaml
module Directions : sig
  type 'a t = { left : 'a; right : 'a; ... }
end
...
  method focus : t option Directions.t
```

I don't think we need the double level of options either, if we want a convenient way to test that all fields are `None`, we  can expose an API like this:

```ocaml
module Directions : sig
  type 'a t = { left : 'a; ... }

  val for_all : 'a t -> f:('a -> bool) -> bool
end
```